tools/webdriver/webdriver/bidi/modules/browsing_context.py

```diff
@@ -178,5 +178,16 @@ def _print(self, result: Mapping[str, Any]) -> Any:
         return result["data"]
 
     @command
-    def set_viewport(self, context: str, viewport: Optional[Mapping[str, Any]] = None) -> Mapping[str, Any]:
-        return {"context": context, "viewport": viewport}
+    def set_viewport(self,
+                     context: str,
+                     viewport: Optional[Mapping[str, Any]] = None,
+                     device_pixel_ratio: Optional[float] = None) -> Mapping[str, Any]:
+        params: MutableMapping[str, Any] = {
+            "context": context,
+            "viewport": viewport,
```

Comment:
Because `viewport` is optional as well, please do the same `None` check. Only `context` is mandatory.